File: src/main/java/kg/mega/kindergarten/services/ChildService.java
package kg.mega.kindergarten.services;

import kg.mega.kindergarten.enums.Delete;
import kg.mega.kindergarten.models.Child;
import kg.mega.kindergarten.models.dtos.ChildSaveDto;
import kg.mega.kindergarten.models.dtos.ChildDto;

import java.util.List;

public interface ChildService {
    ChildDto create(ChildSaveDto childCreateDto);


    ChildDto delete(Long id);

    List<Child> findAllList(int page, int size);

    Child findById(Long id);

    List<Child> findByIdToList(Long childId);

    ChildDto update(Long id, ChildSaveDto childSaveDto, Delete delete);
}

File: src/api/childService.js
import axiosClient from './axiosClient';

const childService = {
  getAll: (page = 0, size = 10) => axiosClient.get('/child/get-list', { params: { page, size } }),
  getById: (id) => axiosClient.get('/child/find-by-id', { params: { id } }),
  create: (childData) => axiosClient.post('/child/create', childData),
  update: (id, childData, deleteStatus = 'ACTIVE') => axiosClient.put('/child/update', childData, { params: { id, delete: deleteStatus } }),
  delete: (id) => axiosClient.delete('/child/delete', { params: { id } }),
};

export default childService;
